services/motion/explore/explore_test.go

```diff
@@ -0,0 +1,343 @@
+package explore
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	"go.viam.com/test"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+const (
+	defaultLinearVelocityMMPerSec    = 200.
+	defaultAngularVelocityDegsPerSec = 60.
+)
+
+type obstacleMetadata struct {
+	position r3.Vector
+	data     int
+	label    string
+}
+
```

Comment:
[note] Tests need to be added to transferal of obstacleDetectors

Reply:
unsure what you mean by "transferal of obstacleDetectors" here